README.md

```diff
@@ -93,9 +93,129 @@ TODO aspect ratio + images alignment + more more images
 
 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu odio gravida, tristique quam quis, dignissim purus. Sed sed neque facilisis, venenatis odio in, dignissim risus. Nulla facilisi. Aliquam dictum volutpat ligula. Quisque vehicula condimentum bibendum. Morbi posuere, libero ac porttitor molestie, sem enim molestie sapien, at consectetur metus lacus nec justo. Sed sollicitudin nisl ut tellus posuere pharetra. Phasellus in rutrum elit. Nunc dui dui, ultricies eu nunc in, dictum gravida eros. Integer fermentum in turpis non ultricies. Cras sit amet sagittis sapien. Integer dignissim mauris ac magna dapibus, non ultrices risus rhoncus. Sed gravida hendrerit mattis. Pellentesque a congue massa. Nullam in cursus libero. Ut ac tristique mauris.
 
+
 # Features
 
-< features > 
+### 🧩 The Nabla Core Profile
+
+Nabla exposes [a curated set of Vulkan extensions and features](https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/src/nbl/video/vulkan/profiles/NablaCore.json) compatible across the GPUs we aim to support. (TODO: on which platforms?)
+
+### 🧩 Physical Device Selection and Filteration
+
+Nabla allows you to select the best GPU for your compute or graphics workload.
+
+```c++
+nbl::video::SPhysicalDeviceFilter deviceFilter = {};
+deviceFilter.minApiVersion = { 1,3,0 };
+deviceFilter.minConformanceVersion = {1,3,0,0};
+deviceFilter.requiredFeatures.rayQuery = true;
+// TODO: add something else here
+deviceFilter(physicalDevices);
+```
+
+### 🧩 SPIR-V and Vulkan as First-Class Citizens
+
+Nabla treats SPIR-V and Vulkan as core components, this ensures full control over [TODO]
+
+### Integration of Renderdoc
+
+Built-in support for capturing frames and debugging with Renderdoc.
+
+```c++
+const IQueue::SSubmitInfo submitInfo = {
+    .waitSemaphores = {},
+    .commandBuffers = {&cmdbufInfo,1},
+    .signalSemaphores = {&signalInfo,1}
+};
+m_api->startCapture(); // Start Renderdoc Capture
+queue->submit({&submitInfo,1});
+m_api->endCapture(); // End Renderdoc Capture
+```
+
+### 🧩 Nabla Event Handler: Seamless GPU-CPU Synchronization
+
+Nabla Event Handler's extensive usage of [Timeline Semaphores]() enables CPU Callbacks on GPU conditions.
+
+You can enqueue callbacks that trigger upon specific GPU conditions, enabling tasks like resource deallocation to be handled only after the GPU has completed the relevant work.
+```c++
+// This doesn't actually free the memory from the pool, the memory is queued up to be freed only after the `scratchSemaphore` reaches a value a future submit will signal
+memory_pool->deallocate(&offset,&size,nextSubmit.getFutureScratchSemaphore());
```

Comment:
I'd probably provide a more longform code example with any semaphore, and not `getFutureScratchSemaphore`

We can do the SIntendedNextSubmit as a separate feature actually (and IUtilities)

Reply:
Hmm not sure what example or code I would use that shows the usages as good as the deallocation thing. Any suggestions? 
---
Yes I have IUtilities in mind to be another point, I suggest mentioning it after or before the asset converter point, what do you think?

Reply:
before, also would make sense to explain SIntendedSubmit within it